### PR TITLE
Add projects and cursor design

### PR DESCRIPTION
**Pull request by KerrynEer**
Add Projects Section - a slideshow of projects
Add Sparkles following cursor around.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -12,17 +12,234 @@
...
-  greetingContainer.innerText = greeting;
+// PROJECT SLIDESHOW LOGIC
+let slideIndex = 1;
+showSlides(slideIndex);
+
+function plusSlides(n) {
+  showSlides(slideIndex += n);
+}
+
+function currentSlide(n) {
```
Comment (lingxz) [inline portfolio/src/main/webapp/script.js]: maybe name this setCurrentSlide?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -12,17 +12,234 @@
...
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+// PROJECT SLIDESHOW LOGIC
+let slideIndex = 1;
+showSlides(slideIndex);
+
+function plusSlides(n) {
```
Comment (lingxz) [inline portfolio/src/main/webapp/script.js]: sounds like changeSlide will be a better name

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -12,17 +12,234 @@
...
+  let i;
+  let slides = document.getElementsByClassName("project");
+  let dots = document.getElementsByClassName("dot");
+
+  if (n > slides.length) {
+      slideIndex = 1;
+  }
+  if (n < 1) {
+      slideIndex = slides.length;
+  }
```
Comment (lingxz) [inline portfolio/src/main/webapp/script.js]: I think it's better semantically if checking whether n is out of range is handled in changeSlide/plusSlide. showSlide is a function that sounds like it should just handle the display, whereas changeSlide function handles these edge cases logic. Having this kind of side effects in showSlide may be confusing in the future.

Comment (KerrynEer) [reply portfolio/src/main/webapp/script.js]: Ah yes good point here! Will make the change 👍

**Review by lingxz — APPROVED** [review]

**Pull request merged**